source/renderer/app/components/notifications/GenericNotification.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
+import { intlShape } from 'react-intl';
+import classNames from 'classnames';
+import Action from '../../actions/lib/Action';
+import NotificationMessage from '../widgets/NotificationMessage';
+import successIcon from '../../assets/images/success-small.inline.svg';
+import spinnerIcon from '../../assets/images/spinner-dark.inline.svg';
+import type { Props as NotificationMessageProps } from '../widgets/NotificationMessage';
+import styles from './GenericNotification.scss';
+
+/**
+ *
+ * Extra props from NotificationMessageProps.js:
+ *
+ * {
+ *   show: boolean,
+ *   children?: Node,
+ *   clickToClose?: boolean,
+ *   hasCloseButton?: boolean,
+ *   order?: 'auto' | number | 'initial' | 'inherit',
+ * }
+ */
+
+type Props = {
+  ...$Exact<NotificationMessageProps>,
+  id: string,
+  icon?: 'success' | 'spinner' | string,
+  iconStyle?: Object,
+  duration?: number,
+  actionToListenAndOpen?: Action<any>,
+  actionToListenAndClose?: Action<any>,
+  openNotification: Action<any>,
+  closeNotification: Action<any>,
+  hasEllipsis?: boolean,
+};
+
+@observer
+export default class GenericNotification extends Component<Props> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  constructor(props: Props) {
```

Comment:
Are you sure this works correctly when only setting up the listeners in the `constructor`?
From what i know the constructor is only called once for each component and not again if it unmounts and then mounts again. You should probably use `componentDidMount` instead

Reply:
Are you sure? It's working pretty well, after unmounting and mounting again.